virttest/qemu_devices/qcontainer.py

```diff
@@ -3201,3 +3202,41 @@ def numa_hmat_cache_define_by_params(self, nodeid, params):
                     aobject=aobject, backend='hmat_type'))
 
         return hmat_cache_devs
+
+    def secret_object(self, secret_id, data=None, secret_format='raw',
+                      encrypt=True, inline=True, secret_file=None):
+        """
+        Generate secret object device with given data, will create a new file
+        when inline is False and file is None
+
+        :param secret_id: the id of the secret object
+        :param data: password, could be None when file is provided and
+                     inline is True
+        :param secret_format: raw or base64
+        :param encrypt: encrypt data when True and format is not raw
```

Comment:
Does the `secret_format` indicates whether it is encrypted or not  already ?
I mean, when using `format=base64` in qemu command, the `data` is encrypted already, is it ?

Reply:
@yanan-fu I suppose the user could give the plain text in data, and `encrypt=True` to let the function help encrypt it to `base64`, or it could give directly `base64` format data and `encrypt=False`, so the usage could be:
`secret_object(secret_id, data='123456', format='base64', encrypt=True)` or
`secret_object(secret_id, data='MTIzNDU2', format='base64', encrypt=False)`
What do you think?

Reply:
Ah~ i only suppose user define the plain text data, and qemu command line use encrypted data that handled by this function, the scenario one indeed.
No storage idea with the scenario two that user define a encrypted data. Call for more idea with other uses that use secret object too  @zhencliu could you have a look ?  Thanks!

Reply:
The secret object used for storage access fully depends on user input, i.e.  user can represent secret data in two ways: 'plain text data(data) + raw(format)' or 'data in base64(data) + base64(format)', we never encrypt/decrypt the sensitive data in VT.  Simply, show me the data and its format, I use it.

Also, for storage access, usually we have no choice for using key-secret(inline passwd) or password-secret(a secret file should exist, containing the passwd), because  only one is supported. 

For this patch, it seems raw/base64 password and key-secret/password-secret are all OK, we can even allow user to pass a file containing password, that's very flexible, but consider devices.secret_object('id'), this is correct from grammer perspective, but it might not make sense. 
What about 'def secret_object(self, secret_id, data, secret_format='raw', inline=True)', data is required, adding data encryption seems not part of responsibility for this function.

Reply:
> The secret object used for storage access fully depends on user input, i.e. user can represent secret data in two ways: 'plain text data(data) + raw(format)' or 'data in base64(data) + base64(format)', we never encrypt/decrypt the sensitive data in VT. Simply, show me the data and its format, I use it.
> 
> Also, for storage access, usually we have no choice for using key-secret(inline passwd) or password-secret(a secret file should exist, containing the passwd), because only one is supported.
> 
> For this patch, it seems raw/base64 password and key-secret/password-secret are all OK, we can even allow user to pass a file containing password, that's very flexible, but consider devices.secret_object('id'), this is correct from grammer perspective, but it might not make sense.

Thanks @zhencliu Good point.

> What about 'def secret_object(self, secret_id, data, secret_format='raw', inline=True)', data is required, adding data encryption seems not part of responsibility for this function.

I am okay with both ways, even it is not flexible enough but it is still sufficient for any type of users, what's your opinion, @yanan-fu ?

Reply:
Agree, I am ok with this too, function is more concise while can meet the different type too.